Sources/_StringProcessing/Engine/MEBuiltins.swift

```diff
@@ -367,4 +546,87 @@ extension String {
     }
     return next
   }
+
+  @inline(never)
```

Comment:
I'll think over how much code duplication we can avoid here later (and otherwise skip these implementation functions in the review). If we're implementing this much duplication, we'll want to make sure we have very, very solid testing in place.